crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -333,53 +335,63 @@ fn build_new_replicated_loglet_configuration(
     preferred_sequencer: Option<NodeId>,
 ) -> Option<ReplicatedLogletParams> {
     let mut rng = thread_rng();
-    // todo make min nodeset size configurable, respect roles, respect StorageState, etc.
-    let nodes_config = metadata().nodes_config_ref();
+    // todo: make the default nodeset size configurable
+    let replication = previous_configuration
+        .map(|config| config.replication.clone())
+        .unwrap_or_else(|| ReplicationProperty::new(NonZeroU8::new(2).expect("to be valid")));
 
-    let log_servers: Vec<_> = observed_cluster_state
-        .alive_nodes
-        .values()
-        .filter(|node| {
-            nodes_config
-                .find_node_by_id(**node)
-                .ok()
-                .is_some_and(|config| config.has_role(Role::LogServer))
-        })
-        .collect();
+    // todo: make nodeset selection strategy configurable
+    let strategy = NodeSetSelectionStrategy::FaultTolerantAdaptive;
+
+    let nodes_config = metadata().nodes_config_ref();
```

Comment:
If was pass in the `NodesConfiguration` as a parameter instead, then we are a bit more functional which could simplify the test (no need for the `TestCoreEnv`.